sklearn/linear_model/tests/test_huber.py

```diff
@@ -0,0 +1,174 @@
+from math import exp
+
+import numpy as np
+from scipy import optimize, sparse
+
+from sklearn.utils.testing import assert_equal
+from sklearn.utils.testing import assert_almost_equal
+from sklearn.utils.testing import assert_array_almost_equal
+from sklearn.utils.testing import assert_greater
+
+from sklearn.datasets import make_regression
+from sklearn.linear_model import HuberRegressor, Ridge, SGDRegressor
+from sklearn.linear_model.huber import _huber_loss_and_gradient
+
+rng = np.random.RandomState(0)
+
+def make_regression_with_outliers(n_samples=50, n_features=20):
+    X, y = make_regression(
+        n_samples=n_samples, n_features=n_features,
+        random_state=0, noise=10.0)
+
+    # Replace 10% of the sample with noise.
+    num_noise = int(0.1 * n_samples)
+    random_samples = rng.randint(0, n_samples, num_noise)
+    X[random_samples, :] = rng.normal(0, 1, (num_noise, X.shape[1]))
+    return X, y
+
+
+def test_huber_equals_ridge_for_high_epsilon():
+    X, y = make_regression_with_outliers()
+    ridge = Ridge(fit_intercept=True, random_state=0, alpha=0.0)
+    ridge.fit(X, y)
+    huber = HuberRegressor(fit_intercept=True, epsilon=10000.0, alpha=0.0)
+    huber.fit(X, y)
+    assert_almost_equal(huber.coef_, ridge.coef_, 1)
+
+
+def test_huber_gradient():
+    """Test that the gradient calculated by _huber_loss_and_gradient is correct"""
+
+    X, y = make_regression_with_outliers()
+    loss_func = lambda x, *args: _huber_loss_and_gradient(x, *args)[0]
+    grad_func = lambda x, *args: _huber_loss_and_gradient(x, *args)[1]
+
+    # Check using optimize.check_grad that the gradients are equal.
+    for i in range(5):
+
+        # Check for both fit_intercept and otherwise.
+        for n_features in [X.shape[1] + 1, X.shape[1] + 2]:
+            w = rng.randn(n_features)
+            w[-1] = np.abs(w[-1])
+            grad_same = optimize.check_grad(
+                loss_func, grad_func, w, X, y, 0.01, 0.1)
```

Comment:
eps here is the default optimize.check_grad argument right? or 0.1? Having trouble reading this